airflow/bin/airflow

```diff
@@ -1,9 +1,30 @@
 #!/usr/bin/env python
 import logging
 import os
+import socket
+import requests
 from airflow import configuration
 from airflow.bin.cli import CLIFactory
 
+def get_private_ip(name=''):
+    r = requests.get("http://169.254.169.254/latest/meta-data/local-ipv4")
```

Comment:
Wish `boto3` had a nice wrapper around this, but https://github.com/boto/boto3/issues/313 has been open for a while. There is `boto.utils.get_instance_metadata()` though:

```python
>>> import boto.utils
>>> instance_metadata = boto.utils.get_instance_metadata()
>>> print instance_metadata['local-ipv4']
10.0.24.67
```
But that is a dependency to another library, which may be available though.

Overall looks like the hostname the node returns:
```
PRODUCTION: amalakar@etl-production-iad-d532cf46:~$ hostname
etl-production-iad-d532cf46
```
Doesn't seem to have a dns entry or an entry in `/etc/hosts`
```
PRODUCTION: amalakar@etl-production-iad-d532cf46:~$ host etl-production-iad-d532cf46
Host etl-production-iad-d532cf46 not found: 3(NXDOMAIN)
```

I have seen lot of systems do depend on the `gethostname` to resolve correctly on a DNS lookup. For example hadoop cli would fail when this is not satisfied. I think any instance that gets provisioned should have these pieces working which is an assumption for lot of systems out there. 

@astahlman should we create a ticket against #provisioning to take care of this? We may end up patching/hacking startup script of many other systems otherwise.